app/models/vim_performance_analysis.rb

```diff
@@ -439,6 +439,107 @@ def self.find_perf_for_time_period(obj, interval_name, options = {})
                   .select(options[:select])
   end
 
+  def self.collect_metrics(obj, start_time, end_time, interval)
+    client = ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::CaptureContext.new(
```

Comment:
Why is a general model like `VimPerformanceAnalysis` dealing with specific providers?